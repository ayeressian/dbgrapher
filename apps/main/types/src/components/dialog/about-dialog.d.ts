import { CSSResultGroup, TemplateResult } from "lit";
import { DBGElement } from "../dbg-element";
export default class extends DBGElement {
    #private;
    private open;
    static get styles(): CSSResultGroup;
    render(): TemplateResult;
    connectedCallback(): void;
}
