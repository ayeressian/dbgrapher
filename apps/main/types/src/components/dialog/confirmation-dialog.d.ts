import { CSSResultGroup, TemplateResult } from "lit";
import { DBGElement } from "../dbg-element";
export default class ConfirmationDialog extends DBGElement {
    #private;
    private static instance;
    private static resultResolve;
    private static result?;
    open: boolean;
    message: string;
    confirmText: string;
    cancelText: string;
    static get styles(): CSSResultGroup;
    render(): TemplateResult;
    constructor();
    static confirm(message: string, confirmText?: string, cancelText?: string): Promise<boolean>;
}
