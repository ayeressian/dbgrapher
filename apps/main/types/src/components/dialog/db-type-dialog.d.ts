import { TemplateResult } from "lit";
import { DBGElement } from "../dbg-element";
export default class extends DBGElement {
    #private;
    private open;
    private selectedDbType;
    connectedCallback(): void;
    render(): TemplateResult;
}
