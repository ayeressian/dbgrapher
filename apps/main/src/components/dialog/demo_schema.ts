import DbGrapherSchema from "../../db-grapher-schema";

const demoSchema: DbGrapherSchema = {
  dbGrapher: { type: "Generic" },
  tables: [
    {
      pos: { x: 1282, y: 853 },
      name: "ram_model",
      columns: [
        { name: "id", type: "int", pk: true },
        { name: "make", type: "string", nn: true },
        { name: "model", type: "string", uq: true, nn: true },
        { name: "speed", type: "int", pk: false, uq: true, nn: true },
        { name: "type", type: "ENUM('ddr4', 'ddr5')", nn: true },
      ],
    },
    {
      pos: { x: 830.78, y: 583.72 },
      name: "cpu_model",
      columns: [
        { name: "id", type: "int", pk: true },
        { name: "make", type: "string", nn: true },
        { name: "model", type: "string", pk: false, nn: true, uq: true },
        { name: "base_clock", type: "int", pk: false, uq: false, nn: true },
        { name: "boost_clock", type: "int", uq: false, nn: false },
        { name: "l1_cache", type: "int", nn: true },
        { name: "l2_cache", type: "int", nn: true },
      ],
    },
    {
      pos: { x: 815.94, y: 247.26 },
      name: "motherboard_model",
      columns: [
        { name: "id", type: "int", pk: true },
        { name: "make", type: "string", nn: true },
        { name: "model", type: "string", nn: true },
        { name: "socket", type: "string", nn: true },
        { name: "ram_type", type: "ENUM('ddr4', 'ddr5')", nn: true },
        { name: "chipset", type: "string", nn: true },
      ],
    },
    {
      pos: { x: 1270.44, y: 310.98 },
      name: "pc",
      columns: [
        { name: "id", type: "int", pk: true },
        {
          name: "fk_cpu_model_id",
          uq: true,
          nn: true,
          fk: { table: "cpu_model", column: "id" },
        },
        {
          name: "fk_psu_model_id",
          uq: true,
          nn: true,
          fk: { table: "psu_model", column: "id" },
        },
        {
          name: "fk_motherboard_model_id",
          uq: true,
          nn: true,
          fk: { table: "motherboard_model", column: "id" },
        },
      ],
    },
    {
      pos: { x: 1277.5, y: 558.9 },
      name: "pc_ram",
      columns: [
        { name: "id", type: "int", pk: true },
        {
          name: "fk_pc_id",
          uq: false,
          nn: true,
          fk: { table: "pc", column: "id" },
        },
        {
          name: "fk_ram_model_id",
          uq: false,
          nn: true,
          fk: { table: "ram_model", column: "id" },
        },
      ],
    },
    {
      pos: { x: 1700.78, y: 305.16 },
      name: "storage",
      columns: [
        { name: "id", type: "int", pk: true },
        {
          name: "fk_pc_id",
          uq: false,
          nn: true,
          fk: { table: "pc", column: "id" },
          pk: false,
        },
        {
          name: "fk_storage_model_id",
          uq: false,
          nn: true,
          fk: { table: "storage_model", column: "id" },
          pk: false,
        },
      ],
    },
    {
      pos: { x: 1609.44, y: 20.32 },
      name: "psu_model",
      columns: [
        { name: "id", type: "int", pk: true },
        { name: "make", type: "string", nn: true },
        { name: "model", type: "string", nn: true },
        { name: "power", type: "int", nn: true },
      ],
    },
    {
      pos: { x: 1263.86, y: 31.59 },
      name: "video_card",
      columns: [
        { name: "id", type: "int", pk: true },
        {
          name: "fk_pc_id",
          uq: false,
          nn: true,
          fk: { table: "pc", column: "id" },
        },
        {
          name: "fk_video_card_model_id",
          uq: false,
          nn: true,
          fk: { table: "video_card_model", column: "id" },
        },
      ],
    },
    {
      pos: { x: 1264.6, y: -204.82 },
      name: "video_card_model",
      columns: [
        { name: "id", type: "int", pk: true },
        { name: "make", type: "string", nn: true },
        { name: "model", type: "string", nn: true },
        { name: "memory", type: "int", nn: true },
      ],
    },
    {
      pos: { x: 2045.06, y: 299.78 },
      name: "storage_model",
      columns: [
        { name: "id", type: "int", pk: true },
        { name: "make", type: "string", nn: true },
        { name: "model", type: "string", nn: true },
        { name: "capacity", type: "int", pk: false, nn: true },
        { name: "type", type: "ENUM('SSD', 'HDD')", nn: true },
      ],
    },
  ],
};
export default demoSchema;
