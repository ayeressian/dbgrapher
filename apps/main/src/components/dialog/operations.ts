import { CSSResultGroup, TemplateResult, css, html } from "lit";
import { customElement } from "lit/decorators.js";
import { DBGElement } from "../dbg-element";

@customElement("dbg-dialog-operations")
export default class extends DBGElement {
  static get styles(): CSSResultGroup {
    return css`
      .operations {
        display: flex;
        flex-wrap: wrap;
        flex-direction: horizontal;
      }
    `;
  }

  render(): TemplateResult {
    return html` <div class="operations"><slot></slot></div> `;
  }
}
