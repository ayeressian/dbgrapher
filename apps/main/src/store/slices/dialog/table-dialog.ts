import { createSlice } from "@reduxjs/toolkit";
import { Point } from "db-viewer";

type TableDialogData = {
  open: boolean;
  tableName?: string;
  cords?: Point;
};

const slice = createSlice({
  initialState: {
    open: false,
  } as TableDialogData,
  name: "tableDialog",
  reducers: {
    close: (): TableDialogData => ({
      tableName: undefined,
      open: false,
      cords: undefined,
    }),
    openCreate: (_, { payload: cords }): TableDialogData => ({
      tableName: undefined,
      open: true,
      cords: cords as Point,
    }),
    openEdit: (_, { payload: tableName }): TableDialogData => ({
      tableName: tableName as string,
      open: true,
      cords: undefined,
    }),
  },
});

export default slice;

export const { reducer, actions } = slice;
