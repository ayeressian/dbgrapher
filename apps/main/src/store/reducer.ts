import { combineReducers, Action } from "redux";
import { reducer as createCords } from "./slices/create-cords";
import { reducer as tableDialog } from "./slices/dialog/table-dialog";
import { reducer as dbViewerMode } from "./slices/db-viewer-mode";
import { reducer as fileDialog } from "./slices/dialog/file-dialog";
import { reducer as fileOpenChooserDialog } from "./slices/dialog/file-open-chooser-dialog";
import { reducer as schema } from "./slices/schema";
import { reducer as loadSchema } from "./slices/load-schema";
import { reducer as loadScreen } from "./slices/load-screen";
import { reducer as cloud } from "./slices/cloud";
import { reducer as localization } from "./slices/localization";
import { reducer as dialogs } from "./slices/dialog/dialogs";
import { reducer as dbTypeDialog } from "./slices/dialog/db-type-dialog";
import { reducer as showUnsavedWarning } from "./slices/show-unsaved-warning";
import { reducer as hint } from "./slices/hint";

const appReducer = combineReducers({
  createCords,
  dbViewerMode,
  cloud,
  hint,
  dialog: combineReducers({
    dialogs,
    dbTypeDialog,
    fileDialog,
    tableDialog,
    fileOpenChooserDialog,
  }),
  loadScreen,
  loadSchema,
  schema,
  localization,
  showUnsavedWarning,
});

export type AppState = ReturnType<typeof appReducer>;

// TODO: change unknown with proper type
const rootReducer = (state: unknown, action: Action): AppState => {
  if (action.type === "RESET") state = undefined;

  return appReducer(state as AppState, action);
};

export default rootReducer;
