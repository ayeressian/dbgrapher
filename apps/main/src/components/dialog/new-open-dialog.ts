import { html, TemplateResult } from "lit";
import store from "../../store/store";
import {
  actions as dialogActions,
  DialogTypes,
} from "../../store/slices/dialog/dialogs";
import fileSvg from "../../../asset/file.svg";
import folderOpenSvg from "../../../asset/folder-open.svg";
import demoSvg from "../../../asset/demo.svg";
import { t } from "../../localization";
import { DBGElement } from "../dbg-element";
import { newFile, openFile } from "../operations";
import { customElement, state } from "lit/decorators.js";
import { actions as schemaAction } from "../../store/slices/schema";
import { actions as laodSchemaAction } from "../../store/slices/load-schema";
import demoSchema from "./demo_schema";

@customElement("dbg-new-open-dialog")
export default class extends DBGElement {
  @state()
  private open = store.getState().dialog.dialogs.newOpenDialog;

  connectedCallback(): void {
    super.connectedCallback();

    this.subscribe(
      (state) => state.dialog.dialogs.newOpenDialog,
      (open) => {
        this.open = open;
      }
    );
  }

  render(): TemplateResult {
    return html`
      <dbg-dialog
        ?show=${this.open}
        centerTitle=${t((l) => l.dialog.fileOpenChooser.welcome)}
      >
        <div slot="body">
          <dbg-dialog-operations>
            <dbg-dialog-operation
              data-testid="new-file"
              @dbg-click=${this.#newFile}
              text=${t((l) => l.dialog.newOpen.operation.newSchema)}
              icon=${fileSvg}
            ></dbg-dialog-operation>
            <dbg-dialog-operation
              data-testid="open-file"
              @dbg-click=${this.#openFile}
              text=${t((l) => l.dialog.newOpen.operation.openSchema)}
              icon=${folderOpenSvg}
            ></dbg-dialog-operation>
            <dbg-dialog-operation
              data-testid="new-file"
              @dbg-click=${this.#demo}
              text=${t((l) => l.dialog.newOpen.operation.demoSchema)}
              icon=${demoSvg}
            ></dbg-dialog-operation>
          </dbg-dialog-operations>
        </div>
      </dbg-dialog>
    `;
  }

  #newFile = (): void => {
    store.dispatch(dialogActions.close(DialogTypes.NewOpenDialog));
    newFile();
  };

  #openFile = (): void => {
    openFile();
  };

  #demo = (): void => {
    store.dispatch(dialogActions.close(DialogTypes.NewOpenDialog));
    store.dispatch(schemaAction.initiate(demoSchema));
    store.dispatch(laodSchemaAction.load());
  };
}
