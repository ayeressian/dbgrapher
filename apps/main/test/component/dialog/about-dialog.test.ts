import "../../../src/components/dialog/about-dialog";
import AboutDialog from "../../../src/components/dialog/about-dialog";
import store from "../../../src/store/store";
import { initComponentTest, removeElement } from "../../helper";
import {
  actions as dialogActions,
  DialogTypes,
} from "../../../src/store/slices/dialog/dialogs";
import { describe, beforeEach, afterEach, it } from "vitest";
import toMatchHtmlSnapshot from "../../to-match-html-snapshot";

describe("dbg-about-dialog", () => {
  let aboutDialog: AboutDialog;

  beforeEach(async () => {
    aboutDialog = await initComponentTest("dbg-about-dialog");
    store.dispatch(dialogActions.open(DialogTypes.AboutDialog));
  });

  afterEach(() => {
    removeElement(aboutDialog);
  });

  it("renders propery", () => {
    toMatchHtmlSnapshot(aboutDialog.shadowRoot?.innerHTML);
  });
});
