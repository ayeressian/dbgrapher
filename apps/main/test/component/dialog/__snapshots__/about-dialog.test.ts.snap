// Vitest Snapshot v1

exports[`dbg-about-dialog > renders propery 1`] = `
"<dbg-dialog centertitle=\\"About\\" showclose=\\"\\" show=\\"\\">
<div slot=\\"body\\">
<p data-testid=\\"text\\">Hello my name is Ara. I initiated this project for my own needs.
If you like relational databases and have knowledge or wish to learn
more about typescript, webcomponent, litelement, svelte, redux, vite,
vitest, playwright and much more you should consider to contribute
to this project.</p>
<p data-testid=\\"footer\\">I hope you enjoy using this application.</p>
</div>
</dbg-dialog>
<style>
p {
width: 500px;
}
</style>"
`;
